syntax = "proto3";

package weather;

service WeatherService {
  rpc StreamWeather (WeatherRequest) returns (stream WeatherResponse);
}

message WeatherRequest {
  string city = 1;
}

message WeatherResponse {
  string city = 1;
  string description = 2;
  float temperature = 3;
  float humidity = 4;
  string timestamp = 5; // Time of weather update
}
